substrate/frame/identity/src/lib.rs

```diff
@@ -1395,16 +1395,21 @@ impl<T: Config> Pallet<T> {
 			},
 		)?;
 
-		// Subs Deposit
-		let new_subs_deposit = SubsOf::<T>::try_mutate(
-			&target,
-			|(current_subs_deposit, subs_of)| -> Result<BalanceOf<T>, DispatchError> {
-				let new_subs_deposit = Self::subs_deposit(subs_of.len() as u32);
-				Self::rejig_deposit(&target, *current_subs_deposit, new_subs_deposit)?;
-				*current_subs_deposit = new_subs_deposit;
-				Ok(new_subs_deposit)
-			},
-		)?;
+		let new_subs_deposit = if SubsOf::<T>::contains_key(&target) {
+			SubsOf::<T>::try_mutate(
+				&target,
+				|(current_subs_deposit, subs_of)| -> Result<BalanceOf<T>, DispatchError> {
+					let new_subs_deposit = Self::subs_deposit(subs_of.len() as u32);
+					Self::rejig_deposit(&target, *current_subs_deposit, new_subs_deposit)?;
+					*current_subs_deposit = new_subs_deposit;
+					Ok(new_subs_deposit)
+				},
+			)?
+		} else {
+			// If the item doesn't exist, there is no "old" deposit, and the new one is zero, so no
+			// need to call rejig, it'd just be zero -> zero.
+			Zero::zero()
```

Comment:
Based on the identity migrator impl, calling `Call::poke_deposit` with a non existing target will never pay any fees -- is this the expected behaviour? I guess it should be fine since the call is gated by root.

Reply:
Yeah it is gated by root, and the only use in the impl of `reap_identity` would return an error if the target did not exist, so it shouldn't end up calling into `poke_deposit` at all.